src/slackmessageparser.ts

```diff
@@ -72,49 +75,52 @@ export class SlackMessageParser {
 			.map((m) => m.trim())
 			.join("\n")
 			.trim();
+
 		// insert @room in place of <!channel> and <!here>
 		rawMessage = rawMessage.replace(/<!channel>/g, "@room");
 		rawMessage = rawMessage.replace(/<!here>/g, "@room");
+
+		// Replace &amp; with literal & - fixes &amp;amp;
+		rawMessage = rawMessage.replace(/&amp;/g, '&');
+
 		let msg = rawMessage;
-		let html = md.render(msg);
-		// insert the colour hacks
-		html = html.replace(/;BEGIN_FONT_COLOR_HACK_(.*?);/g, "<font color=\"$1\">");
-		html = html.replace(/;END_FONT_COLOR_HACK;/g, "</font>");
-		msg = msg.replace(/;BEGIN_FONT_COLOR_HACK_(.*?);/g, "");
-		msg = msg.replace(/;END_FONT_COLOR_HACK;/g, "");
+		let result = null as RegExpExecArray | null;
+
+		// detect slack formatted urls (<scheme:uri|title>)
+		while ((result = /<([a-zA-Z][a-zA-Z0-9+\-.]+:[^\|>]+)\|([^>]*)>/g.exec(msg)) !== null) {
+			// convert slack url to markdown formatted url
+			msg = msg.replace(result[0], '[' + result[2] + '](' + result[1] + ')');
+		}
+
+		let html = msg;
 
 		// replace user mentions
-		let result = null as RegExpExecArray | null;
-		result = /<@([a-zA-Z0-9]*)>/g.exec(msg);
-		while (result !== null) {
+		while ((result = /<@([a-zA-Z0-9]*)>/g.exec(msg)) !== null) {
```

Comment:
tslint will fail at this, as it is an assignment inside of an if/while/etc. (`npm run lint`)